sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventDataAggregator.java

```diff
@@ -36,7 +36,11 @@
 class EventDataAggregator extends FluxOperator<EventData, EventDataBatch> {
     private static final ClientLogger LOGGER = new ClientLogger(EventDataAggregator.class);
 
-    private final AtomicReference<EventDataAggregatorMain> downstreamSubscription = new AtomicReference<>();
+    private volatile Subscription downstreamSubscription;
+    private static final AtomicReferenceFieldUpdater<EventDataAggregator, Subscription> DOWNSTREAM_SUBSCRIPTION
+        = AtomicReferenceFieldUpdater.newUpdater(EventDataAggregator.class, Subscription.class,
+            "downstreamSubscription");
```

Comment:
Just curious; Is there any potential issue if the type of `downstreamSubscription` is declared as `EventDataAggregatorMain`? if `source.subscribe()` auto downcast this type to `Subscription`, then we can avoid type check and casting on every call to `getNumberOfEvents()`

Reply:
Mm, good idea. I only had it as Subscription because `Operators.setOnce` took `Subscription` as a parameter. I can be more specific.